### DISCUSSION TITLE
How to control mpv through neovim?

### DISCUSSION THREAD

**Discussion by lovelindhoni** [mpv-player/mpv#14819] [Q&A]
I am writing some nvim functions that controls the playback of mpv through the IPC connection. I would lke to read some data like the filtered-metadata of a file, but its said in the manual that reading from the terminal output is a bad idea.  Can I get some heads-up on how to use mpv's lua scripting here, get the data from the lua scripts into the context of nvim

Thanks

**Comment by guidocella** [comment]
Not parsing terminal output refers to log messages. You are supposed to read the output of the JSON IPC.

**Comment by lovelindhoni** [comment]
Thanks, I am trying to read the output from the JSON IPC using neovim's luvref bindings. 
```lua
function send_mpv_command()
  local pipe = uv.new_pipe(false)

  pipe:connect(mpv_socket, function(err)
    if err then
      print("Error connecting :", err)
      return
    end

    local command = '{ "command": ["get_property", "metadata"] }\n'
    print("Connected ", mpv_socket)

    pipe:write(command, function(write_err)
      if write_err then
        print("Error writing to socket:", write_err)
      else
        print("Command sent to MPV")
      end

      pipe:shutdown() -- shutdown of the writing side of the pipe

      pipe:read_start(function(read_err, data)
        if read_err then
          print("Error reading socket:", read_err)
        elseif data then
          print(data)
        else
          print("MPV socket closed")
        end
        pipe:close()
      end)
    end)
  end)
end
```


I can't see any data is prevented. It works perfectly well for commands such as `cycle pause` but only the output is not being read from the JSON IPC

**Reply by guidocella** [reply]
I don't know how you would read from the socket that way. In my vim script I just called

```vim
json_decode(system('echo ''{ "command": ["get_property", "foo"] }'' \| socat - /tmp/mpv-socket'))
```
